---
title: "06_proteinPanel_tumor_analysis"
author: "SandraTietscher"
date: "2020-10-26"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

This script looks in more detail into the epithelial cell subset. This is done on the Protein Panel data, because it contains more tumor-relevant markers than the RNA panel.
Given that there are still not many tumor markers in the panel, this is a rather coarse analysis. 

# Load libraries and data

```{r read-libraries-and-data, message=FALSE, results="hide"}
# Load libraries
library(ggplot2)
library(SingleCellExperiment)
library(scater)
library(tidyverse)
library(dittoSeq)
library(viridis)
library(cowplot)
library(ggpubr)
library(scran)
library(dplyr)
library(Seurat)

# Read SingleCellExperiment object (with tumor cells only)
sce_tumor <- readRDS(paste0(getwd(), "/output/ProteinPanel/sce_ProteinPanel_tumor.rds"))

TIG2 <- c("TBB111", "TBB129", "TBB165", "TBB171", "TBB184", "TBB338")
```

# Tumor cell subclustering

## Calculate UMAP on all tumor cells

```{r tumor-umap}
markers_include_umap <- c("CAIX", "Cleaved_CP", "Ki-67", "E-Cadherin", "panCK", "HLA-ABC", "MMP9", "SMA", "CD274")

sce_tumor <- runUMAP(sce_tumor, exprs_values = "exprs", 
                     subset_row = rownames(assay(sce_tumor, "scaled")) %in% markers_include_umap)
```

### Visualize features on UMAP

```{r tumor-umap-features, fig.width=8, fig.height=8}

dittoDimPlot(sce_tumor, var = "sample", reduction.use = "UMAP", size = 0.5, 
                               color.panel = metadata(sce_tumor)$colour_vectors$sample, opacity = 0.2) 

dittoDimPlot(sce_tumor, var = "sample", reduction.use = "UMAP", size = 0.5, 
                               color.panel = metadata(sce_tumor)$colour_vectors$sample, split.by = "sample", opacity = 0.2) 

dittoDimPlot(sce_tumor, var = "TIG", reduction.use = "UMAP", size = 0.5, 
                               color.panel = metadata(sce_tumor)$colour_vectors$TIG)

dittoDimPlot(sce_tumor, var = "TLS", reduction.use = "UMAP", size = 0.5, 
                               color.panel = metadata(sce_tumor)$colour_vectors$TLS)
```

### Visualize marker expression on UMAP

```{r tumor-umap-markerexpression, fig.width=35, fig.height=25, message=FALSE, warning=FALSE}

# Plot expression of all markers
p.list <- list()
for(i in markers_include_umap){
  p.list[[i]] <- plotUMAP(sce_tumor, colour_by = i, by_exprs_values = "scaled", 
                          point_size = 0.5, point_alpha = 1, shape_by = "panel_type")  + scale_colour_gradient2(name = i, low = "#2166ac", mid = "white", high = "#b2182b", limits = c(-3, 3))
}
plot_grid(plotlist = p.list, ncol = 5)
```

## Run graph-based clustering for tumor subset (Phenograph)

Display clusters on UMAP.

```{r tumor-clustering-umap, fig.width=8, fig.height=8}
cluster_counts_tumor <- subset(assay(sce_tumor, "normalized"), rownames(assay(sce_tumor, "normalized")) %in% markers_include_umap)

g <- buildSNNGraph(cluster_counts_tumor, k=50)
clust <- igraph::cluster_louvain(g)$membership
table(clust)
#write.csv(table(clust), paste0(getwd(), "/output/ProteinPanel/tumor_cluster_cellnumbers.csv"), row.names = FALSE)

#add cluster information to sce object
sce_tumor$tumor_cluster <- as.factor(clust)

#plot on umap
dittoDimPlot(sce_tumor, var = "tumor_cluster", reduction.use = "UMAP", size = 0.2, do.label = TRUE, opacity = 0.2)
dittoDimPlot(sce_tumor, var = "tumor_cluster", reduction.use = "UMAP", size = 0.2, split.by = "tumor_cluster", opacity = 0.2)
```

### Plot mean expression for each cluster

```{r tumor-clusters-expression-heatmap, fig.width=5, fig.height=5}
clustermean <- aggregateAcrossCells(sce_tumor, ids = DataFrame(cluster = sce_tumor$tumor_cluster), average = TRUE)
colnames(clustermean) <- clustermean$cluster

# Center and scale raw counts
assay(clustermean, "scaled") <- t(scale(t(assay(clustermean, "counts"))))

# Normalize raw counts
norm_counts <- t(apply(assay(clustermean, "counts"), 1, function(x)(x-min(x))/(max(x)-min(x))))
assay(clustermean, "normalized", withDimnames = FALSE) <- norm_counts

## Plot
# Z-scaled
dittoHeatmap(clustermean[markers_include_umap,], assay = "scaled", scale = "none",
            heatmap.colors = magma(100), 
            cluster_cols = TRUE, show_colnames = TRUE,
            breaks = seq(-3, 3, length.out = 101))

# Normalized (0-1)
dittoHeatmap(clustermean[markers_include_umap,], assay = "normalized", scale = "none",
            heatmap.colors = viridis(100), 
            cluster_cols = TRUE, show_colnames = TRUE)
```

### Plot sample proportions per cluster

```{r sample-proportions-cluster}
props <- as.data.frame(prop.table(table(sce_tumor$tumor_cluster, sce_tumor$sample), margin = 1))
colnames(props) <- c("cluster", "sample", "proportion")

ggplot(props, aes(cluster, y = proportion, fill=sample)) +
  geom_bar(stat="identity")+
  theme(axis.title.x=element_blank(), axis.title.y = element_blank())+
  theme(panel.background = element_blank(), axis.text.x = element_text(angle = 30))
```


# Compare marker expression in TIG2 versus TIG3

Single-cell violin plots.

```{r TIG-marker-comparison-singlecell, fig.width=5, fig.height=6}

exprs <- data.frame(HLA_ABC = assay(sce_tumor, "exprs")["HLA-ABC",],
                    `Ki-67` = assay(sce_tumor, "exprs")["Ki-67",],
                    CAIX = assay(sce_tumor, "exprs")["CAIX",],
                    Cleaved_CP = assay(sce_tumor, "exprs")["Cleaved_CP",],
                    `PD-L1` = assay(sce_tumor, "exprs")["CD274",],
                    MMP9 = assay(sce_tumor, "exprs")["MMP9",])
tumor_tab <- cbind(exprs, colData(sce_tumor))
tumor_tab <- pivot_longer(tumor_tab, 1:6, names_to = "marker", values_to = "exprs")

ggplot(tumor_tab, aes(x=TIG, y = exprs, fill = TIG))+
  geom_violin(aes(fill=TIG))+
  geom_boxplot(position=position_dodge(width=0.9), width=0.1, color="black", aes(fill=TIG))+
  facet_wrap(~marker, scale = "free")+
  theme(axis.line.x = element_line(colour = "black", size = 0.25),
        axis.line.y = element_line(colour = "black", size = 0.25),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        axis.title.x = element_blank())+
  ylab("Marker expression (arcsin-transformed counts)")
```

Comparison of patient means.

```{r TIG-marker-comparison-patientmeans, fig.width=5, fig.height=6, message=FALSE}

# Patient means
counts <- data.frame(HLA_ABC = assay(sce_tumor, "counts")["HLA-ABC",],
                    `Ki-67` = assay(sce_tumor, "counts")["Ki-67",],
                    CAIX = assay(sce_tumor, "counts")["CAIX",],
                    Cleaved_CP = assay(sce_tumor, "counts")["Cleaved_CP",],
                    `PD-L1` = assay(sce_tumor, "counts")["CD274",],
                    MMP9 = assay(sce_tumor, "counts")["MMP9",])
tumor_tab_2 <- cbind(counts, colData(sce_tumor))
tumor_tab_2 <- pivot_longer(tumor_tab_2, 1:6, names_to = "marker", values_to = "counts")

sample_means <- data.frame(sample = levels(factor(tumor_tab_2$sample)))
for (i in c("HLA_ABC", "CAIX", "Cleaved_CP", "Ki.67", "PD.L1", "MMP9")) {
  tumor.i <- filter(tumor_tab_2, marker == i)
  i.means <- tumor.i %>% group_by(sample) %>% dplyr::summarise(mean = mean(counts))
  sample_means[,i] <- i.means$mean
}

sample_means$TIG <- c(rep("TIG2", 6), rep("TIG3", 6))
sample_means <- pivot_longer(sample_means, 2:7, names_to = "marker", values_to = "counts")

ggplot(sample_means, aes(TIG, counts, fill = TIG))+
    geom_boxplot(aes(fill=TIG))+
    geom_point(position=position_dodge(width=0.75), aes(group=TIG))+
    facet_wrap(~marker, scales = "free")+
    stat_compare_means(method = "wilcox.test", aes(label=..p.format..), vjust = 1, label.x.npc = 0.5)+
    theme(axis.line.x = element_line(colour = "black", size = 0.25),
        axis.line.y = element_line(colour = "black", size = 0.25),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, size=1),
        panel.background = element_blank(),
        strip.background = element_blank())
```

## Correlation of HLA-ABC expression (protein level) and HER2 expression (transcript level)

```{r HLA_ABC-HER2-correlation, fig.width=5, fig.height=4, message=FALSE}
# Read in Seurat object
tumor10x <- readRDS("data/RNAseq_epithelial.rds")

# Calculate averages
tumor10x_sub <- tumor10x["ERBB2",]
Idents(tumor10x_sub) <- tumor10x_sub$orig.ident
cluster.averages <- as.data.frame(t(AverageExpression(object = tumor10x_sub, return.seurat = FALSE)[[1]]))
cluster.averages$sample <- rownames(cluster.averages)

# Remove samples that were not measured by IMC
not.measured <- c("TBB011", "TBB330")
cluster.averages <- cluster.averages[!(rownames(cluster.averages) %in% not.measured),]

# Merge with HLA-ABC data from IMC
IMC_means <- sample_means %>% filter(marker == "HLA_ABC")
comb.tab <- merge(IMC_means, cluster.averages, by = "sample")
colnames(comb.tab)[4] <- "HLA_ABC"

ggplot(comb.tab, aes(HLA_ABC, ERBB2)) +
  geom_point(aes(color=TIG))+
  geom_smooth(method = lm, color = "black", se=FALSE)+
  stat_cor(method="spearman")+
  theme(axis.line.x = element_line(colour = "black", size = 0.25),
        axis.line.y = element_line(colour = "black", size = 0.25),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, size=1),
        panel.background = element_blank())
```

# Define cell status 

Based on Ki67, Carbonic Anhydrase IX and Cleaved Caspase/Cleaved PARP expression

## Define Ki-67 positive tumor cells

```{r Ki67-cutoff-cluster, fig.width=6, fig.height=3}
dittoPlot(sce_tumor, "Ki-67", group.by = "tumor_cluster",
    plots = c("vlnplot"), assay = "exprs")+
  geom_hline(yintercept = 0.7, lty = "dashed")
```
```{r Ki67-cutoff-full, fig.width=3, fig.height=4}
dittoPlot(sce_tumor, "Ki-67", group.by = "panel_type",
    plots = c("vlnplot"), assay = "exprs", vlnplot.width = 1)+
  geom_hline(yintercept = 0.7, lty = "dashed")
```

```{r Ki67-status-IHC-IMC, fig.width=3, fig.height=3}
# Assign Ki67-status
sce_tumor$Ki67_status <- ifelse(assay(sce_tumor, "exprs")["Ki-67",] >= 0.7, TRUE, FALSE)

Ki67_props <- as.data.frame(proportions(table(sce_tumor$sample, sce_tumor$Ki67_status), margin = 1))
colnames(Ki67_props) <- c("sample", "status", "IMC_proportion")
Ki67_props <- Ki67_props %>% filter(status == TRUE)

# Read in clinically determined Ki67 percentage (IHC)
Ki67_IHC <- read.csv("data/Ki67_IHC.csv")

Ki67_comb <- merge(Ki67_props, Ki67_IHC, by = "sample")

ggplot(Ki67_comb, aes(IMC_proportion*100, Ki67_IHC)) +
  geom_point()+
  geom_smooth(method = lm, color = "black", se=FALSE)+
  stat_cor(method="spearman")+
  theme(axis.line.x = element_line(colour = "black", size = 0.25),
        axis.line.y = element_line(colour = "black", size = 0.25),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, size=1),
        panel.background = element_blank())+
  xlab("Ki67_IMC [%]")+
  ylab("Ki67_clinical_IHC [%]")
```

## Define hypoxic and apoptotic tumor cells

```{r cutoff-cluster, fig.width=6, fig.height=3}
dittoPlot(sce_tumor, "Cleaved_CP", group.by = "tumor_cluster",
    plots = c("vlnplot"), assay = "exprs")+
  geom_hline(yintercept = 0.5, lty = "dashed")

dittoPlot(sce_tumor, "CAIX", group.by = "tumor_cluster",
    plots = c("vlnplot"), assay = "exprs")+
  geom_hline(yintercept = 0.9, lty = "dashed")
```
```{r cutoff-full, fig.width=3, fig.height=4}
dittoPlot(sce_tumor, "Cleaved_CP", group.by = "panel_type",
    plots = c("vlnplot"), assay = "exprs", vlnplot.width = 1)+
  geom_hline(yintercept = 0.5, lty = "dashed")

dittoPlot(sce_tumor, "CAIX", group.by = "panel_type",
    plots = c("vlnplot"), assay = "exprs", vlnplot.width = 1)+
  geom_hline(yintercept = 0.9, lty = "dashed")
```

```{r hypoxic-apoptotic-status}
# Assign hypoxia-status and apoptotic status
sce_tumor$hypoxic <- ifelse(assay(sce_tumor, "exprs")["CAIX",] >= 0.9, TRUE, FALSE)
sce_tumor$apoptotic <- ifelse(assay(sce_tumor, "exprs")["Cleaved_CP",] >= 0.5, TRUE, FALSE)

# Check overlap
table(sce_tumor$hypoxic, sce_tumor$apoptotic)
table(sce_tumor$hypoxic, sce_tumor$Ki67_status)
table(sce_tumor$apoptotic, sce_tumor$Ki67_status)
```

## Assign cell status for tumor cells and attach to main SCE
  
Around 1/3 of all apoptotic cells are also hypoxic.
Around 1/3 of all hypoxic cells are Ki67+.

Define order of relevance for assigning cell status:  
1. Ki-67  
2. hypoxic  
3. apoptotic  

```{r tumor-cell-status-TIG, fig.width=8, fig.height=5}
sce_tumor$cell_status <- ifelse(sce_tumor$Ki67_status == TRUE, "tumor_Ki67", ifelse(sce_tumor$hypoxic == TRUE, "tumor_hypoxic", ifelse(sce_tumor$apoptotic == TRUE, "tumor_apoptotic", "tumor")))

# Plot proportions
tab <- as.data.frame(prop.table(table(sce_tumor$sample, sce_tumor$cell_status), margin = 1))
colnames(tab) <- c("sample", "cell_status", "proportion")
tab$TIG <- ifelse(tab$sample %in% TIG2, "TIG2", "TIG3")

ggplot(tab, aes(x= TIG, y = proportion, color=TIG))+
  geom_boxplot()+
  geom_point()+
  facet_wrap(~cell_status, scales="free", ncol=4, strip.position="top")+
  theme(axis.ticks.x = element_blank(),
        axis.text.x=element_text(angle = 90, hjust=1),
        axis.title.x = element_blank(),
        panel.background = element_blank(),
        panel.border = element_rect(color = "black", fill = NA, size = 1),
        strip.background = element_blank(),
        legend.background = element_blank())+
  ylab("Of total [%]")+
  stat_compare_means(method = "wilcox.test", aes(label=..p.format..), label.x.npc = 0.5)
```

```{r tumor-cell-status-barplot, fig.width=6, fig.height=4}
ggplot(tab, aes(sample, y = proportion, fill=cell_status)) +
  geom_bar(stat="identity")+
  theme(axis.title.x=element_blank(), axis.title.y = element_blank())+
  theme(panel.background = element_blank(), axis.text.x = element_text(angle = 30))
```

# Update SCE objects

```{r save-object}
saveRDS(sce_tumor, "output/ProteinPanel/sce_ProteinPanel_tumor.rds")
```

## Add tumor cell status to main SCE object as "subtype"

```{r update-main-SCE}
sce_protein <- readRDS(file = "output/ProteinPanel/sce_ProteinPanel_celltypes.rds")
sce_protein$subtype <- sce_protein$celltype

tumor_ids <- colnames(sce_tumor)
sce_protein[,tumor_ids]$subtype <- sce_tumor$cell_status

saveRDS(sce_protein, "output/ProteinPanel/sce_ProteinPanel_subtypes.rds")
```

# Cytomapper visualization

```{r cytomapper-tumor, message = FALSE}
library(cytomapper)

# Load mask files
path.to.images <- "data/masks/ProteinPanel_masks"
all_masks <- loadImages(path.to.images, pattern = "_mask.tiff")
all_masks

# Add image number
imagenr <- read.csv("data/masks/ProteinPanel_masks/image_number_HIER.csv", header=FALSE)[1]
mcols(all_masks)$ImageNumber <- imagenr$V1

# Scale images
all_masks <- scaleImages(all_masks, 2^16-1)
head(unique(as.numeric(all_masks[[1]])))

cur_img <- getImages(all_masks, "20200924_ST_BCexh_H_TBB129_s0_p3_r6_a6_ac_ilastik_s2_Probabilities_mask")

# Plot cells
plotCells(cur_img, object = sce_protein,
            img_id = "ImageNumber", cell_id = "CellNumber",
            colour_by = "celltype")

plotCells(cur_img, object = sce_protein,
            img_id = "ImageNumber", cell_id = "CellNumber",
            colour_by = c("CD11c", "LAMP3", "CD3"), exprs_values = "exprs")

```




